db/seed.js

```diff
@@ -1,39 +1,37 @@
 require('dotenv').config();
-const faker = require('faker');
 const { MongoClient } = require('mongodb');
+const cluster = require('cluster');
+const numCPUs = require('os').cpus().length;
+const generateDocument = require('./generateDocument');
 
 const dbAddress = process.env.DB_ADDRESS || 'localhost';
 const dbName = process.env.DB_NAME || 'marzagat_overview';
 const collectionName = process.env.COLLECTION_NAME || 'restaurants';
 const seedNum = parseInt(process.env.SEED_NUM, 10) || 10000000;
-const batchSize = parseInt(process.env.SEED_BATCH_SIZE, 10) || 5000;
+const batchSize = parseInt(process.env.SEED_BATCH_SIZE, 10) || 15000;
+const schemaIdName = process.env.SCHEMA_ID_NAME || '_id';
 
-const url = `mongodb://${dbAddress}:27017`;
+const connectToDb = async () => {
+  const url = `mongodb://${dbAddress}/`;
+  const client = await MongoClient.connect(url);
+  const collection = client.db(dbName).collection(collectionName);
+  return { client, collection };
+};
 
-const generateDocument = id => (
-  {
-    _id: id,
-    name: faker.commerce.productName(),
-    tagline: faker.lorem.words((id % 4) + 3),
-    type: faker.lorem.word(),
-    vicinity: `${faker.address.streetAddress()}, ${faker.address.city()}`,
-    priceLevel: faker.random.number({ min: 1, max: 4 }),
-    zagatFood: faker.finance.amount(0, 5, 1),
-    zagatDecor: faker.finance.amount(0, 5, 1),
-    zagatService: faker.finance.amount(0, 5, 1),
-    longDescription: faker.lorem.paragraph((id % 2) + 11),
-  }
-);
+const indexDb = async (collection) => {
+  const indexObj = {};
+  indexObj[schemaIdName] = 1;
+  await collection.ensureIndex(indexObj, { unique: true });
+};
 
 const seedBatch = (minId, maxId, collection) => (
   new Promise(async (resolve, reject) => {
     const docs = [];
     for (let i = minId; i < maxId; i++) {
-      docs.push(generateDocument(i));
+      docs.push(generateDocument(i, schemaIdName));
```

Comment:
I'm curious about why you pass schemaIdName as a parameter into generateDocument? It seems like you are assigning it to a variable in the first place in case you want to change the name of the index property so that it can be easily changed, but is there a reason to pass it in as a parameter versus just grabbing it from closure scope? Is it to minimize the function's external dependencies? Since it will always be the same value I was thinking you might not need to pass it in as a param...

Reply:
totally makes sense -- per our convo yesterday I'll update this as well!

Reply:
ah! i know we discussed this yesterday but I actually just realized why I'm passing in schemaIdName as a parameter into `generateDocument()`. it's because I moved generateDocument into a separate file! So it no longer has access to the global constants declared at the top of `seed.js`